### DISCUSSION TITLE
borealis的分支问题

### DISCUSSION THREAD

**Discussion by kelvin8906** [xfangfang/wiliwili#262] [Q&A]
我发现borealis，wiliwili和moonlight都有一个魔改版，要支持deko3d的话，都需要如此魔改吗？
接入新版本borealis从构建都重构工作量好像都超出了想象😅

**Comment by xfangfang** [comment]
chiaki 的那个 borealis 用的是旧版的，新版支持了 yoga 布局，如果要接入 deko3d 那就一定需要新版 borealis了。

moonlight 的 borealis 主要是增加了触摸和更好的列表支持， deko3d 相关的东西相比原版应该变动不大；
wiliwili 的 borealis 也是基于 moonlight 做的，虽然改了很多代码但是大部分都是为了支持其他平台，对于borealis的api变动很少（或者几乎没有，有也是扩展），涉及到 deko3d相关的东西，针对 mpv 的支持做了微调。

所以用原版和另外两个版本，在主要代码上来说应该区别不大（都需要按照新版本 borealis API 重写布局），wiliwili 的 borealis 支持 cmake 可能引入会稍微容易一些，而且支持 chiaki 现在还不支持的 iOS。

**Reply by kelvin8906** [reply]
那我研究一下wiliwili版本，cmake引入仓库里就不维护原版代码了，谢谢解释。

deko3d我看moonlight的非主线分支已经挺完善了，应该主要工作量就是在引入新版borealis然后更新布局

**Reply by zeromake** [reply]
@kelvin8906 moonlight 的 borealis 地址在哪我看看它 nanovg 做了什么

**Reply by zeromake** [reply]
不用了我找到了：https://github.com/XITRIX/borealis https://github.com/Adubbz/nanovg-deko3d

**Comment by kelvin8906** [comment]
我跟moonlight的作者尝试了deko感觉在串流没办法搞，还没gl绘图快。感谢方方的信息呀，我现在build版本也是用你的PKG。如果方方也玩chiaki可以看看这个硬解版

https://git.sr.ht/~kkwong/chiaki/

**Reply by xfangfang** [reply]
这样呀，我还以为deko3d会快一些呢，感谢分享

**Comment by xfangfang** [comment]
@kelvin8906 
参考：https://github.com/Egoistically/chiaki 对switch版的颜色映射做了一点调整，确实感觉颜色更加正确了一些，可以也试试看，我感觉效果还不错。

```diff
diff --git a/switch/src/io.cpp b/switch/src/io.cpp
index eb44880..9872d6e 100644
--- a/switch/src/io.cpp
+++ b/switch/src/io.cpp
@@ -48,8 +48,8 @@ void main()
 		(texture(plane3, uv_var).r - (16.0 / 255.0)) / ((240.0 - 16.0) / 255.0) - 0.5);
 	vec3 rgb = mat3(
 		1.0,		1.0,		1.0,
-		0.0,		-0.21482,	2.12798,
-		1.28033,	-0.38059,	0.0) * yuv;
+		0.0,		-0.18733,	1.85563,
+		1.57480,	-0.46812, 	0.0) * yuv;
 	out_color = vec4(rgb, 1.0);
 }
 )glsl";
@@ -73,8 +73,8 @@ void main()
 	);
 	vec3 rgb = mat3(
 		1.0,		1.0,		1.0,
-		0.0,		-0.21482,	2.12798,
-		1.28033,	-0.38059,	0.0) * yuv;
+		0.0,		-0.18733,	1.85563,
+		1.57480,	-0.46812, 	0.0) * yuv;
 	out_color = vec4(rgb, 1.0);
 }
 )glsl";

```

这里是一些相关的讨论：https://github.com/Egoistically/chiaki/commit/0f12a639af52e2d294c7d1e22bd504bcb5f7fe8c#comments

修改前：
![IMG_7419](https://github.com/xfangfang/wiliwili/assets/16631584/9f9e87b4-0afa-4522-9c15-5a91647ca32c)

修改后：
![IMG_7420](https://github.com/xfangfang/wiliwili/assets/16631584/08443e71-2fe5-4f59-8fb5-a240cdbbf882)

**Reply by kelvin8906** [reply]
刚把老头环DLC打完😁马上加上了
https://git.sr.ht/~kkwong/chiaki/refs/v2.2.0-patch3-color-adjust
